packages/core/src/types/Vector.ts

```diff
@@ -270,7 +316,34 @@ export class Vector2 implements Type {
     return {x: this.x, y: this.y};
   }
 
+  /**
+   * Check if two vectors are exactly equal to each other.
+   *
+   * @remarks
+   * If you need to compensate for floating point inaccuracies, use the
+   * {@link equals} method, instead.
+   *
+   * @param other - The vector to compare.
+   */
   public exactlyEquals(other: Vector2): boolean {
     return this.x === other.x && this.y === other.y;
   }
+
+  /**
+   * Check if two vectors are equal to each other.
+   *
+   * @remarks
+   * This method allows passing an allowed error margin when comparing vectors
+   * to compensate for floating point inaccuracies. To check if two vectors are
+   * exactly equal, use the {@link exactlyEquals} method, instead.
+   *
+   * @param other - The vector to compare.
+   * @param epsilon - The allowed error threshold when comparing the vectors.
+   */
+  public equals(other: Vector2, epsilon = 0.000001): boolean {
+    return (
+      Math.abs(this.x - other.x) <= epsilon + Number.EPSILON &&
+      Math.abs(this.y - other.y) <= epsilon + Number.EPSILON
+    );
+  }
```

Comment:
Why is it implemented differently than `Matrix2D.equals()`?

Reply:
If anything, `Matrix2D.equals()` should be updated to work this way as well. The reason I'm adding `Number.EPSILON` is to account for cases like this

```ts
const a = new Vector2(2.5, 2.5);
const b = new Vector2(2.4, 2.4);

a.equals(b, 0.1);
// => false
```

because in magical floating point land

```
2.5 - 2.4 = 0.10000000000000009
```

I can make the same change to `Matrix2D.equals()` in this PR as well. Or I can open a new PR with the change if you'd prefer that.

Reply:
You can update it in this PR
Also make sure the arguments are called the same thing (`threshold` vs `epsilon`) and that the default value uses the same constant.

Reply:
Done 👍